linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala

```diff
@@ -232,6 +268,68 @@ private[emitter] final class VarGen(jsGen: JSGen, nameGen: NameGen,
     Ident(avoidClashWithGlobalRef("$" + field + "_" + subField), origName)
   }
 
+  private def maybeFromModule(className: ClassName, ident: Ident)(
+      implicit moduleContext: ModuleContext, globalKnowledge: GlobalKnowledge,
+      pos: Position): Tree = {
+    val targetModule = globalKnowledge.getModule(className)
+    if (targetModule == moduleContext.id) {
+      VarRef(ident)
+    } else {
+      DotSelect(VarRef(envModuleFieldIdent(targetModule.id)), ident)
+    }
+  }
+
+  private def maybeExport(ident: Ident, tree: Tree, mutable: Boolean = false)(
+      implicit moduleContext: ModuleContext, pos: Position): Tree = {
+    if (moduleContext.leaf) {
+      tree
+    } else {
+      config.moduleKind match {
+        case ModuleKind.NoModule =>
+          throw new AssertionError("non-leaf module in NoModule mode")
+
+        case ModuleKind.ESModule =>
+          val export = Export((ident -> ExportName(ident.name)) :: Nil)
+          Block(tree, export)
+
+        case ModuleKind.CommonJSModule =>
+          val exportsVarRef = VarRef(Ident("exports"))
+          val name = StringLiteral(ident.name)
+
+          if (mutable) {
+            // defineProperty method
+            val defProp =
+              genIdentBracketSelect(VarRef(Ident("Object")), "defineProperty")
+
+            val getterDef = {
+              StringLiteral("get") -> Function(arrow = false, Nil, {
+                Return(VarRef(ident))
+              })
+            }
+
+            val setterDef = {
+              val x = Ident("x")
+              StringLiteral("set") -> Function(arrow = false, List(ParamDef(x, rest = false)), {
+                Assign(VarRef(ident), VarRef(x))
+              })
+            }
+
+            // Options passed to the defineProperty method
+            val descriptor = ObjectConstr(
+              getterDef ::
+                (StringLiteral("configurable") -> BooleanLiteral(true)) ::
+                Nil
+            )
+
+            Block(tree, Apply(defProp, List(exportsVarRef, name, descriptor)))
```

Comment:
Consider refactoring as:
```suggestion
            val export = Apply(defProp, List(exportsVarRef, name, descriptor))
            Block(tree, export)
```
and then you can factor out `val export =` and `Block(tree, export)` outside of the `if (mutable)`.